indico/modules/events/registration/controllers/management/sections.py

```diff
@@ -0,0 +1,92 @@
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from flask import request, jsonify
+
+from indico.core.db import db
+from indico.modules.events.registration.controllers.management import RHManageRegFormBase
+from indico.modules.events.registration.models.items import RegistrationFormSection
+from indico.modules.events.registration.models.registration_form_fields import (RegistrationFormField,
+                                                                                RegistrationFormFieldData)
+
+
+class RHManageRegFormSectionBase(RHManageRegFormBase):
+    CSRF_ENABLED = False
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.section
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageRegFormBase._checkParams(self, params)
+        self.section = RegistrationFormSection.get_one(request.view_args['section_id'])
+
+
+class RHRegFormAddSection(RHManageRegFormBase):
+    CSRF_ENABLED = False
+
+    def _process(self):
+        section = RegistrationFormSection(registration_form=self.regform)
+        section.title = request.json['title']
+        section.description = request.json.get('description', '')
+        db.session.add(section)
+        db.session.flush()
+        return jsonify(section.angular_data)
+
+
+class RHRegFormToggleSectionState(RHManageRegFormSectionBase):
+    def _process(self):
+        enable = request.args.get('enable')
+        if enable == 'true':
+            self.section.enabled = True
+        else:
+            self.section.enabled = False
+        db.session.flush()
+        return jsonify(self.section.angular_data)
+
+
+class RHRegFormDeleteSection(RHManageRegFormSectionBase):
+    def _process(self):
+        self.section.is_deleted = True
+        db.session.flush()
+        return jsonify(success=True)
+
+
+class RHRegFormAddField(RHManageRegFormSectionBase):
+    def _process(self):
+        field_data = request.json['fieldData']
+        form_field = RegistrationFormField(parent_id=self.section.id, registration_form=self.regform)
+        form_field.title = field_data.pop('caption')
+        form_field.description = field_data.pop('description', '')
+        form_field.enabled = not field_data.pop('disabled')
+        db.session.add(form_field)
+        db.session.flush()
+        db.session.add(RegistrationFormFieldData(field_id=form_field.id, data=field_data))
```

Comment:
If you associate it with the field via the relationship you don't even have to add it to the session